host/fs/fs.go

```diff
@@ -94,3 +104,27 @@ var (
 	inhibited bool
 	used      bool
 )
+
+func translateOpMIPS(op uint) (uint, error) {
+	// Decode the arm/x64 encoding and reencode as MIPS specific linux ioctl.
+	// arm/x64: DIR(2), SIZE(14), TYPE(8), NR(8)
+	// mips:    DIR(3), SIZE(13), TYPE(8), NR(8)
+	// Check for size overflow.
+	if (op & (1 << (13 + 8 + 8))) != 0 {
+		return 0, errors.New("fs: op code size is too large")
+	}
+	const mask = (1 << (13 + 8 + 8)) - 1
+	out := op & mask
+	// Convert dir.
+	switch op >> (14 + 8 + 8) {
+	case 0: // none
+		out |= 1 << (13 + 8 + 8)
+	case 1: // write
+		out |= 2 << (13 + 8 + 8)
+	case 2: // read
+		out |= 3 << (13 + 8 + 8)
```

Comment:
I believe, this is wrong. It should be
```go
case 1: // write
	out |= 4 << (13 + 8 + 8)
case 2: // read
	out |= 2 << (13 + 8 + 8)
```

Reply:
Oh you're right, sorry about that. Pushed a fixed version.